src/components/common/ColonyDecisions/DecisionDialog/DecisionDialog.tsx

```diff
@@ -48,27 +48,27 @@ const validationSchema = object()
 export type DecisionDialogValues = InferType<typeof validationSchema>;
 
 export interface DecisionDialogProps extends DialogProps {
+  decision?: Decision;
   ethDomainId?: number;
-  handleSubmit?: (values: DecisionDialogValues) => void;
 }
 
 const DecisionDialog = ({
   cancel,
   close,
+  decision,
   ethDomainId,
-  handleSubmit,
 }: DecisionDialogProps) => {
   const { user } = useAppContext();
   const { pathname } = useLocation();
   const { editor } = useRichTextEditor();
+  const dispatch = useDispatch();
   const navigate = useNavigate();
   const domainId = getDomainId(ethDomainId);
   const walletAddress = user?.walletAddress || '';
-  const draftDecision = getDecisionFromLocalStorage(walletAddress);
-  const content = draftDecision?.description;
+  const content = decision?.description;
 
   const handleSubmitDialog = (values: DecisionDialogValues) => {
-    handleSubmit?.(values);
+    dispatch({ type: ActionTypes.DECISION_DRAFT_CREATED, payload: values });
```

Comment:
Dispatch an action to the store when we create a decision. This tells redux to store the decision so that it's available to the entire app.

Reply:
I noticed you are going away from the pattern of using action creators - is that intentional?

Reply:
No, thank you for pointing out. I have created a couple for these